jsonrpc/types.go

```diff
@@ -104,6 +104,20 @@ type block struct {
 	Uncles          []types.Hash        `json:"uncles"`
 }
 
+func (b *block) Copy() *block {
```

Comment:
Do we need copy transactions and uncles as well, via copy() and allocating new slices?

Reply:
And bloom filter

Reply:
Yep, its in a TODO in the code.

Reply:
Ok, so this `Clone` function is there because `types.Header` also had one. But two things, in this case, I do not think `Clone` is totally necessary (I do not want to remove it though since I do not remember why was there). Second, the block is copied only on filter updates which do never have transactions. So, to me we can keep it like this.
Otherwise, we would have to complicate it a little bit creating also a `Clone` function for `jsonrpc.Transaction`.